File: trunk/scripts/expand_alias.pl
#!/usr/bin/perl
use strict;	   # insist that all variables be declared
use diagnostics;   # expand the cryptic warnings
undef $/;          # read one whole file in one scalar

MAIN:{

  # Replace the word passed as input with the line in the alias file
  # which defines that word as an alias.
  # For example, if "pp" was the input to this script, this
  # script will print the line
  # alias pp=<the name of the alias>
  # provided that this alias exists.
  
  my $in = $ARGV[0];
  
  open(FILE, "<$ENV{HOME}/.bash_aliases");
  my $text = <FILE>;
  close(FILE);

  my @lines = split("\n", $text);

  my $match = 0;
  foreach my $line (@lines){
    next unless ($line =~ /(alias $in=.*?)$/);
    $match = 1;
    print "$line\n";
    last;
  }

  if (!match){
    print "$in\n";
  }
}

File: projects/python/expand_alias.py
#!/usr/bin/python

import sys
import os
import re # perl-style regular expressions

def expandAliases(aliasName):
    
   HOME     = os.getenv("HOME")
   fileName = HOME + '/.bash_aliases'

   FILE = open(fileName, "r")
   data = FILE.read()
   FILE.close()
   
   lines = data.split("\n")
   
   out = ""
   
   for line in lines:

       matches = re.match('^(alias ' + aliasName + '=.*?)$', line); # raw match
       if matches:
           out = matches.group(1)
       # end if
    
       if out == "":
           out = aliasName
       #end if

   #end for
   
   out = re.sub('^alias', 'a', out);
   
   print out

# end expandAliases()

# Main program
numArgs   = len(sys.argv)
if numArgs >= 2 and sys.argv[1] == "a":
    aliasName = sys.argv[numArgs-1]
    expandAliases(aliasName)
# end if
